stores/wallet.ts

```
@@ -88,14 +177,335 @@ export const useWalletStore = defineStore('wallet', () => {
+    magicDIDToken?: string
+  }) {
+    let tempAccountId = generateAccountIdFromWalletAddress(walletAddress)
+    try {
+      await fetchUserRegisterCheck({ accountId: tempAccountId })
+    } catch (error) {
+      if (error instanceof FetchError && error.data?.error === 'USER_ALREADY_EXIST') {
+        tempAccountId = error.data.alternative as string
+      } else {
+        throw error
+      }
+    }
+
+    const startTime = Date.now()
+    let hasError = false
+    const payload = {
+      accountId: tempAccountId,
+      email: prefilledEmail
+    }
+
+    // Loop until registration is successful, user cancels or timeout
+    do {
+      // Check if registration time exceeds the limit
+      if (Date.now() - startTime > REGISTER_TIME_LIMIT_IN_TS) {
+        throw createError({
+          statusCode: 408,
+          data: { description: $t('account_register_timeout') }
+        })
+      }
+
+      try {
+        // Skip registration modal if email is provided
+        if (!payload.email || hasError) {
+          type ModalResult = { accountId: string; email: string; displayName?: string }
+          let modalResult: ModalResult | null = null
+          // Close login panel first to avoid focus trap
+          bookstoreApiStore.closeLoginPanel()
+          await nextTick()
+          await new Promise<void>((resolve) => {
+            modal.open(RegistrationModal, {
+              email: payload?.email,
+              accountId: payload?.accountId,
+              isAccountIdHidden: true,
+              isDisplayNameHidden: true,
+              onSubmit: (data: ModalResult) => {
+                modalResult = data
+                modal.close()
+                resolve()
+              },
+              onClose: () => {
+                modal.close()
+                resolve()
+              }
+            })
+          })
+          if (!modalResult) {
+            // User canceled the registration, return false immediately
+            return false
```

very hard to trace code if we have 3 ways to exit this loop. Maybe we should stick with `break` `continue` `throw error` instead of direct return